### DISCUSSION TITLE
How to extract MIKE3 initial and boundary conditions from HYCOM 3D uv, salinity and temperature data?

### DISCUSSION THREAD

**Discussion by mecwes** [DHI/mikeio#576] [Q&A]
Hi,
I've got one netcdf file with HYCOM data with 3d variables in it: u, v, salinity and temperature and 2d variable 'water level'. Also the water depth is stored in that file. Timestep is 3hr, and 40 depth layers, 23 lon values and 20 lat values (resolution 0.08 deg).
Using xarray, I can do all sorts of interpolations in time, depth and geographic space,

Furthermore, I have a mesh of my MIKE3 model and an empty 3D dfsu file with all the items needed as a base to create initial conditions and boundary conditions. I'm able to find the boundary points in themes and interpolate the hycom data to these points. This procedure worked well for a MIKE21 boundary and initial conditions.

So in theory, it should be easy to go from HYCOM structure to MIKE3 structure, create a mike dataarray, combine into dataset and export as dfsu file.
However, I don't know the depths of the depth sigma layers in this file and am a bit stuck in creating the 3d mike dataarrays with the right dimensions, time axis, vertical layers, and access the values in those 3d dataarrays. Could someone point me out to how to work this out further?

I'm almost there, but miss a little guidance to make the last steps.

Thanks,

Kind regards,
Marco

**Comment by ecomodeller** [comment]
Boundaries to 3d models are a bit of work to get right, are you aware of https://www.mikepoweredbydhi.com/products/mike-cloud/mike-data-link which handles this for you?

**Reply by mecwes** [reply]
Hi, 
I'm aware of the data-link service. The idea is good and have tried it for two days but came along some problems/issues that could not be solved:
1- datalink uses the daily hycom data (I think). after comparison with the original 3hrly hycom data large differences were found between the data link data and the original data. So in terms of accuracy, I'm not a big fan.
2 - During the trial period, the service hang for a few days, couldn't do anything with it such as canceling the request. So, the service isn't reliable enough for me. 
3 - In order to automate things, the data link is not a good solution. It would be nice if one could extract data through an api.
4 - The data was not available for the period I needed.

Given the above, I have to create these 3d initial and boundary conditions with my own scripts.


Regards,
Marco

**Reply by nielsriegels** [reply]
Data Link uses 3-hour HYCOM data. HYCOM data are available from November 2003 to the present.

**Reply by sudheesh-k** [reply]
Hi
Can you please share it if you don't mind.

Regards
Sudheesh
